tests/bluetooth/bsim_bt/bsim_test_rpa_central/src/tester.c

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "bs_bt_utils.h"
+#include "zephyr/bluetooth/addr.h"
+#include "zephyr/bluetooth/bluetooth.h"
+#include "zephyr/bluetooth/conn.h"
+#include "zephyr/toolchain/gcc.h"
+#include <host/hci_core.h>
+
+#include <stdint.h>
+#include <string.h>
+
+DEFINE_FLAG(flag_new_address);
+
+static struct bt_le_adv_param params;
+static struct bt_le_ext_adv_start_param start_params;
+static struct bt_le_ext_adv *adv;
+
+void scanned_cb(struct bt_le_ext_adv *adv, struct bt_le_ext_adv_scanned_info *info)
+{
+	static bool init;
+	static bt_addr_le_t old_addr, new_addr;
+	static uint64_t old_time, new_time, diff;
```

Comment:
```suggestion
	static bt_addr_le_t old_addr;
	bt_addr_le_t new_addr;
	static int64_t old_time;
	int64_t new_time, diff;
```

Not all of these should be `static`

The uptime functions return `int64_t`

`new_time` and `diff` can also be declared in the `if` statement to reduce scope